src/Evolution/Systems/GrMhd/ValenciaDivClean/PalenzuelaEtAl.cpp

```diff
@@ -0,0 +1,173 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/PalenzuelaEtAl.hpp"
+
+#include <cmath>
+#include <limits>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/PrimitiveRecoveryData.hpp"
+#include "NumericalAlgorithms/RootFinding/TOMS748.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Overloader.hpp"
+
+// IWYU pragma: no_forward_declare EquationsOfState::EquationOfState
+
+/// \cond
+namespace grmhd {
+namespace ValenciaDivClean {
+namespace PrimitiveRecoverySchemes {
+
+namespace {
+template <size_t ThermodynamicDim>
+class FunctionOfX {
```

Comment:
I would've found a short comment `// Note: the variable x is defined in the documentation` useful since `x` is usually a spatial coordinate.